sdk/cosmos/azure_data_cosmos/src/handler/retry_handler.rs

```diff
@@ -0,0 +1,141 @@
+use crate::retry_policies::{RetryPolicy, RetryResult};
+use async_trait::async_trait;
+use azure_core::http::{request::Request, RawResponse};
+use typespec_client_core::async_runtime::get_async_runtime;
+
+// Helper trait to conditionally require Send on non-WASM targets
+#[cfg(not(target_arch = "wasm32"))]
+pub trait ConditionalSend: Send {}
+#[cfg(not(target_arch = "wasm32"))]
+impl<T: Send> ConditionalSend for T {}
+
+#[cfg(target_arch = "wasm32")]
+pub trait ConditionalSend {}
+#[cfg(target_arch = "wasm32")]
+impl<T> ConditionalSend for T {}
+
+/// Trait defining the interface for retry handlers in Cosmos DB operations
+///
+/// This trait provides a contract for implementing retry logic that wraps HTTP requests
+/// with automatic retry capabilities. Implementations can inject custom retry policies
+/// and handle both transient failures (errors) and non-success HTTP responses.
+#[allow(dead_code)]
+#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
+pub trait RetryHandler: Send + Sync {
+    /// Sends an HTTP request with automatic retry logic
+    ///
+    /// This method wraps the provided sender callback with retry logic, automatically
+    /// handling transient failures and implementing exponential backoff. The method
+    /// will continue retrying until either:
+    /// - The request succeeds (non-error 2xx status)
+    /// - The retry policy determines no more retries should be attempted
+    /// - Maximum retry attempts are exceeded
+    ///
+    /// # Arguments
+    /// * `request` - Mutable reference to the HTTP request to send (may be modified by retry policy)
+    /// * `sender` - Callback function that performs the actual HTTP request. This function
+    ///              takes a mutable request reference and returns a future that resolves to
+    ///              a `RawResponse` or error.
+    ///
+    /// # Type Parameters
+    /// * `Sender` - Function type that takes `&mut Request` and returns a Future
+    /// * `Fut` - Future type returned by the sender that resolves to `Result<RawResponse>`
+    ///
+    /// # Returns
+    /// `Result<RawResponse>` - The final response (success or failure after all retry attempts)
+    async fn send<Sender, Fut>(
+        &self,
+        request: &mut Request,
+        sender: Sender,
+    ) -> azure_core::Result<RawResponse>
+    where
+        Sender: Fn(&mut Request) -> Fut + Send + Sync,
+        Fut: std::future::Future<Output = azure_core::Result<RawResponse>> + ConditionalSend;
+}
+
+/// Concrete retry handler implementation with exponential back off.
+/// This handler provides automatic retry capabilities for Cosmos DB operations using
+/// a pluggable retry policy system. It wraps HTTP requests with intelligent retry logic
+/// that handles both transient network errors and HTTP error responses.
+#[derive(Debug, Clone)]
+pub struct BackOffRetryHandler;
```

Comment:
_**non-blocking**_: In Rust, I think this would more often just be a few bare functions in a module, rather than grouping it into a type, since there's no state. In .NET/Java, types are the only place to put functions, so having a "zero-size" type just to hold some methods happens a lot more often than in Rust.

**However, there's really no need to churn this PR to change that. I just wanted to point it out for the future.**